spec/models/product_drive_spec.rb

```diff
@@ -35,8 +36,13 @@
   end
 
   it "calculates and returns all donated organization item quantities by name and date" do
-    create(:donation, :with_items, item_quantity: 2, product_drive: product_drive, issued_at: '26-01-2023')
+    # an organization must have an item to be able to list it with item_quantities_by_name_and_date
+    item = create(:item, organization: organization)
+
+    create(:donation, :with_items, item: item, item_quantity: 2, product_drive: product_drive, issued_at: '26-01-2023')
+
     result = product_drive.item_quantities_by_name_and_date(Time.zone.parse('23/01/2023')..Time.zone.parse('26/01/2023'))
+
     expect(result.length).to eq product_drive.organization.items.count
     expect(result.count(2)).to eq(1)
     expect(result.count(0)).to eq(product_drive.organization.items.count - 1)
```

Comment:
should these be hard-coded values?

Reply:
Yes, if at all possible. In this case there _should_ only be one item.